src/Utils/Path.hs

```diff
@@ -0,0 +1,140 @@
+module Utils.Path
+  ( computeRootPath
+  , computeAbsoluteImportPath
+  , computeTargetPath
+  , mergePaths
+  , PathUtils(..)
+  )
+where
+
+import           System.FilePath                (dropFileName, makeRelative, replaceExtension, dropTrailingPathSeparator, takeDirectory,  splitPath
+                                                , joinPath
+                                                , splitFileName
+                                                )
+import           System.Directory               (doesFileExist,  doesDirectoryExist
+                                                )
+import qualified PackageDotJSON
+import qualified MadlibDotJSON
+import Debug.Trace (trace)
+import Data.List (isPrefixOf, (\\))
+
+
+data PathUtils = PathUtils { readFile :: FilePath -> IO String, canonicalizePath :: FilePath -> IO FilePath, getCurrentDirectory :: IO FilePath }
```

Comment:
A wrapper for all IO stuff that needs to be injected in tests in order to control the environment.